package.json

```
@@ -17880,6 +17880,15 @@
 						},
 						"when": "gitlens:plus:state == 6"
 					},
+					{
+						"id": "pro-expired-reactivate",
+						"title": "Reactivate Pro Power-up",
+						"description": "Reactivate your Pro account and experience all the new [Pro features](https://gitkraken.com/gitlens/pro-features?utm_source=gitlens-extension&utm_medium=in-app-links) and the full [GitKraken DevEx platform](https://gitkraken.com/devex?utm_source=gitlens-extension&utm_medium=in-app-links).\n\n[Reactivate Pro](command:gitlens.plus.upgrade?%7B%22source%22%3A%22walkthrough%22%7D)\n\n**Pro Features**\n$(gitlens-graph)  [Commit Graph](command:gitlens.openWalkthrough?%7B%22step%22%3A%22visualize%22,%22source%22%3A%22walkthrough%22%7D) — visualize your repository and keep track of all work in progress\n$(rocket)  [Launchpad](command:gitlens.openWalkthrough?%7B%22step%22%3A%22launchpad%22,%22source%22%3A%22walkthrough%22%7D) — stay focused and keep your team unblocked\n$(gitlens-code-suggestion)  [Code Suggest](command:gitlens.openWalkthrough?%7B%22step%22%3A%22code-collab%22,%22source%22%3A%22walkthrough%22%7D) — free your code reviews from unnecessary restrictions\n$(gitlens-cloud-patch)  [Cloud Patches](command:gitlens.openWalkthrough?%7B%22step%22%3A%22code-collab%22,%22source%22%3A%22walkthrough%22%7D) — easily and securely share code with your teammates\n$(gitlens-worktrees-view)  **Worktrees** — work on multiple branches simultaneously\n$(gitlens-workspaces-view)  **Workspaces** — group and manage multiple repositories together\n$(graph-scatter)  [Visual File History](command:gitlens.openWalkthrough?%7B%22step%22%3A%22visualize%22,%22source%22%3A%22walkthrough%22%7D) — visualize the evolution of a file and quickly identify when the most impactful changes were made and by whom",
+						"media": {
+							"markdown": "walkthroughs/welcome/pro-reactivate.md"
+						},
+						"when": "gitlens:plus:state == 6"
```

```suggestion
						"when": "gitlens:plus:state == 7"
```

You will want to do a search for `gitlens:plus:state` across `package.json` and find all cases where we need to account for the new state you have made (7). For example, you'll need a new addition for text content in worktrees view.

---

In addition, this will cause a new value to be sent out in telemetry so we will want to make sure all consumers of GL telemetry are set up to handle and understand the new state, and the difference between "free" and "paidexpired" if the distinction is important.

---

@axosoft-ramint 
> In addition, this will cause a new value to be sent out in telemetry so we will want to make sure all consumers of GL telemetry are set up to handle and understand the new state, and the difference between "free" and "paidexpired" if the distinction is important.

What changes do we need to do in order to have this? Do I need to update something else in the GitLens side or is this related exclusively to other projects consuming GL telemetry? Thanks!

---

@sergiolms I don't think we need to change anything in the telemetry, just that we should communicate it so that the projects, filters, views, etc. that use telemetry are aware of the new state and can adjust appropriately.